salt/metalk8s/addons/nginx-ingress-control-plane/control-plane-ip.jinja

```diff
@@ -0,0 +1,29 @@
+{# This whole block is used to "know" the Ingress external IP used by Dex.
+  It will be removed once we can have a known LoadBalancer IP for Ingress. #}
+{% if '_errors' in pillar.metalk8s.nodes %}
+  {# Assume this is the bootstrap Node and we haven't an apiserver yet #}
+  {%- set bootstrap_id = grains.id %}
+{%- elif pillar.metalk8s.nodes | length <= 1 %}
+  {# Only one node (or even, zero) can/should only happen during bootstrap #}
+  {%- set bootstrap_id = grains.id %}
+{%- else %}
+  {%- set bootstrap_id = none %}
+  {%- for minion_id, node in pillar.metalk8s.nodes.items() %}
+    {%- if 'bootstrap' in node.roles %}
+      {%- set bootstrap_id = minion_id %}
+      {%- break %}
+    {%- endif %}
+  {%- endfor %}
```

Comment:
This loop could be rewritten with something that we already have (forgot about it):

```jinja
{%- set bootstrap_nodes = salt.metalk8s.minions_by_role('bootstrap') %}
{%- if bootstrap_nodes %}
  {%- set bootstrap_id = bootstrap_nodes | first %}
{%- endif %}
```